build_r.R

```diff
@@ -7,76 +7,63 @@
 
 # R returns FALSE (not a non-zero exit code) if a file copy operation
 # breaks. Let's fix that
-.handle_result <- function(res){
-    if (!res){
-        stop("Copying files failed!")
-    }
+.handle_result <- function(res) {
+  if (!res) {
+    stop("Copying files failed!")
+  }
 }
 
 # Make a new temporary folder to work in
 unlink(x = "lightgbm_r", recursive = TRUE)
 dir.create("lightgbm_r")
 
 # copy in the relevant files
-result <- file.copy(
-    from = "R-package/./"
-    , to = "lightgbm_r/"
-    , recursive = TRUE
-    , overwrite = TRUE
-)
+result <- file.copy(from = "R-package/./",
+                    to = "lightgbm_r/",
+                    recursive = TRUE,
+                    overwrite = TRUE)
 .handle_result(result)
 
-result <- file.copy(
-    from = "include/"
-    , to = file.path("lightgbm_r", "src/")
-    , recursive = TRUE
-    , overwrite = TRUE
-)
+result <- file.copy(from = "include/",
+                    to = file.path("lightgbm_r", "src/"),
+                    recursive = TRUE,
+                    overwrite = TRUE)
 .handle_result(result)
 
-result <- file.copy(
-    from = "src/"
-    , to = file.path("lightgbm_r", "src/")
-    , recursive = TRUE
-    , overwrite = TRUE
-)
+result <- file.copy(from = "src/",
+                    to = file.path("lightgbm_r", "src/"),
+                    recursive = TRUE,
+                    overwrite = TRUE)
 .handle_result(result)
 
-result <- file.copy(
-    from = "CMakeLists.txt"
-    , to = file.path("lightgbm_r", "inst", "bin/")
-    , recursive = TRUE
-    , overwrite = TRUE
-)
+result <- file.copy(from = "CMakeLists.txt",
+                    to = file.path("lightgbm_r", "inst", "bin/"),
+                    overwrite = TRUE)
```

Comment:
is this what was causing the problem? (looks like you removed the `recursive = TRUE`)

A lot of these are formatting changes, so trying to find whichever thing was causing the issue

Reply:
I had two errors:

* on `devtools::document(pkg = "lightgbm_r/")` (because of the trailing slash)
* on the file with recursive (it causes a warning, which causes an error when R is setup to throw an error on warnings)